src/OrchardCore/OrchardCore.ContentLocalization.Abstractions/OrchardHelperExtensions.cs

```diff
@@ -21,7 +22,11 @@ public async static Task<CultureInfo> GetContentCultureAsync(this IOrchardHelper
             var contentManager = orchardHelper.HttpContext.RequestServices.GetService<IContentManager>();
             var cultureAspect = await contentManager.PopulateAspectAsync(contentItem, new CultureAspect());
 
-            return cultureAspect.Culture;
+            // Workaround to ignore the default culture that has been set to CultureAspect.Culture
+            // for the content item which doesn't have a LocalizationPart attached
+            return contentItem.Has<LocalizationPart>()
```

Comment:
Here we end up with the same problem. We don't want to know about any part from here, it's the PopulateAspectAsync that should take care of returning the proper CultureAspect.

Reply:
Here in this Method we would need to be able to say default the value to CultureInfo.CurrentUICulture or not.

Reply:
Ok I can live with this for now but we need to find something better than looking for a specific ContentPart. Maybe more something like look for any ContentPart that inherits ILocalizable.

Reply:
Ok I can live with this for now but we need to find something better than looking for a specific ContentPart. Maybe more something like look for any ContentPart that inherits ILocalizable.

Reply:
And also, I think it will break some Views where we look for a culture for defining RTL or LTR. Because now it could return a null value. That's mainly why I was not fixing it here.

Reply:
Did you check my comment [here](https://github.com/OrchardCMS/OrchardCore/pull/7307#discussion_r508064029)